### ISSUE TITLE
Add simple streaming API

### ISSUE THREAD

**Issue by clue** [clue/php-buzz-react#46]
An experimental streaming API has been implemented via #31. However, it's quite cumbersome to work with.

In particular, one still has to wait on a promise progress event until one can access a stream object.

Instead, we should add a simple API that returns a stream right away and then emits events on it once the HTTP response comes in.

**Comment by clue** [comment]
Also take connection hijacking into account for protocols that write outgoing data even after receiving the response headers.

**Comment by clue** [comment]
This feature will likely be resolved with #30. API draft:

```
$browser->get($url)->then(function (Response $response) {
    $stream = $response->stream();

    $stream->on('data', function ($chunk) {
        echo $chunk;
    });
});
```

**Comment by clue** [comment]
I've played around with several approaches to a sane streaming API while implementing this via #56:

* The response could implement the ReadableStreamInterface. This works out reasonably well for the receiving side through the decorator pattern, but does not really fit for outgoing requests.
* The fetch API offers additional methods and requires a custom Response object which still has to implement the base ResponseInterface. This works out reasonably well for the receiving side through the decorator pattern, but does not really fit for outgoing requests.
* The fetch API resolves as soon as the headers has been received without waiting for the whole body. This is the main takeaway here.
